src/txTransformers.ts

```diff
@@ -96,10 +96,19 @@ const transformTxOutput = (output: TransactionOutput): TransactionOutput => {
   }
 }
 
+/**
+ * If the auxiliary data hash does not match the provided auxiliary data, it is replaced with the
+ * correct hash. This is useful if the auxiliary data was reserialized canonically.
+ */
 const transformAuxiliaryDataHash = (
+  auxiliaryDataHash:
+    | FixlenBuffer<typeof AUXILIARY_DATA_HASH_LENGTH>
+    | undefined,
   auxiliaryData: unknown | undefined,
 ): FixlenBuffer<typeof AUXILIARY_DATA_HASH_LENGTH> | undefined =>
-  auxiliaryData ? blake2b256(encodeToCbor(auxiliaryData)) : undefined
+  auxiliaryData === undefined
```

Comment:
I'd add a comment similar to: `Some clients don't pass in auxiliary data along with the auxiliary data hash. Thus if auxiliary data is not provided, return the original hash (otherwise we would replace the provided hash with a hash of undefined data).`

Reply:
Added